core/src/main/java/org/mobilitydata/gtfsvalidator/validator/ValidationContext.java

```diff
@@ -0,0 +1,41 @@
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.auto.value.AutoValue;
+import java.time.ZonedDateTime;
+import org.mobilitydata.gtfsvalidator.input.GtfsFeedName;
+
+/**
+ * A read-only context passed to particular validator objects. It gives information relevant for
+ * validation: properties of the feed as a whole, system properties (current time) etc.
+ */
+@AutoValue
+public abstract class ValidationContext {
+  public static Builder builder() {
+    return new AutoValue_ValidationContext.Builder();
+  }
+
+  /** Represents a name of a GTFS feed, such as "nl-openov". */
+  public abstract GtfsFeedName feedName();
+
+  /**
+   * The time when validation started.
+   *
+   * <p>Validator code should use this property instead of calling LocalDate.now() etc. for the
+   * following reasons:
+   *
+   * <ul>
+   *   <li>current date and time is changing but it should not randomly affect validation notices;
```

Comment:
Super valid points.
🗨️  Reminds me 3D rendering loops having multiple methods to serve time
https://docs.unity3d.com/Manual/TimeFrameManagement.html 🗨️